src/poetry/plugins/application_plugin.py

```diff
@@ -7,14 +7,20 @@
 
 if TYPE_CHECKING:
     from poetry.console.application import Application
+    from poetry.console.commands.command import Command
 
 
 class ApplicationPlugin(BasePlugin):
     """
-    Base class for plugins.
+    Base class for application plugins.
     """
 
-    type = "application.plugin"
+    group = "poetry.application.plugin"
+
+    @property
+    def commands(self) -> list[type[Command]]:
```

Comment:
Should the `AddCommandPlugin` in `test_application` return something in `commands()`?